pkg/server/api/agent/v1/service_test.go

```diff
@@ -2,17 +2,217 @@ package agent_test
 
 import (
 	"context"
+	"errors"
 	"testing"
 
+	"github.com/sirupsen/logrus"
 	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/pkg/common/telemetry"
 	"github.com/spiffe/spire/pkg/server/api/agent/v1"
 	"github.com/spiffe/spire/pkg/server/api/rpccontext"
 	agentpb "github.com/spiffe/spire/proto/spire-next/api/server/agent/v1"
+	"github.com/spiffe/spire/proto/spire-next/types"
+	"github.com/spiffe/spire/proto/spire/common"
+	"github.com/spiffe/spire/proto/spire/server/datastore"
 	"github.com/spiffe/spire/test/fakes/fakedatastore"
 	"github.com/spiffe/spire/test/spiretest"
+	"github.com/stretchr/testify/require"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
+func TestBanAgent(t *testing.T) {
+	agentTrustDomain := "example.org"
+	agentPath := "spire/agent/agent-1"
+
+	for _, tt := range []struct {
+		name            string
+		reqID           *types.SPIFFEID
+		dsError         error
+		expectedErr     error
+		expectedLogMsgs []spiretest.LogEntry
+	}{
+		{
+			name: "Ban agent succeeds",
```

Comment:
is it possible to check DS contents in these tests? (just an idea)

Reply:
I believe it is already being done? At line 203, the datastore RPC `FetchAttestedNode` is called to check the agent has really been banned.